service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java

```diff
@@ -180,6 +184,60 @@ protected VesselTransportMeansEntity getVesselTransportMeansForRelocation(FaCatc
         return faCatch.getFishingActivity().getVesselTransportMeans().iterator().next();
     }
 
+    protected SizeDistribution getSizeDistribution(FaCatchEntity faCatch) {
+        if (faCatch == null) {
+            return null;
+        }
+
+        SizeDistribution result = new SizeDistribution();
+        if (faCatch.getSizeDistributionCategoryCode() != null && faCatch.getSizeDistributionCategoryCodeListId() != null) {
+            CodeType code = new CodeType();
+            code.setValue(faCatch.getSizeDistributionCategoryCode());
+            code.setListID(faCatch.getSizeDistributionCategoryCodeListId());
+        }
+
+        if (faCatch.getSizeDistributionClassCode() != null && faCatch.getSizeDistributionClassCodeListId() != null) {
+            CodeType code = new CodeType();
+            code.setValue(faCatch.getSizeDistributionClassCode());
+            code.setListID(faCatch.getSizeDistributionClassCodeListId());
+            result.setClassCodes(Lists.newArrayList(code));
+        }
+
+        return result;
+    }
+
+    protected String getSizeDistributionCategoryCode(SizeDistribution sizeDistribution) {
+        if (sizeDistribution == null || sizeDistribution.getCategoryCode() == null) {
+            return null;
+        }
+
+        return sizeDistribution.getCategoryCode().getValue();
+    }
+
+    protected String getSizeDistributionCategoryCodeListId(SizeDistribution sizeDistribution) {
+        if (sizeDistribution == null || sizeDistribution.getCategoryCode() == null) {
+            return null;
+        }
+
+        return sizeDistribution.getCategoryCode().getListID();
+    }
+
+    protected String getSizeDistributionClassCode(SizeDistribution sizeDistribution) {
+        if (sizeDistribution == null || sizeDistribution.getClassCodes() == null || sizeDistribution.getClassCodes().isEmpty()) {
+            return null;
+        }
+
+        return sizeDistribution.getClassCodes().get(0).getValue();
```

Comment:
We should get the approach confirmed by Andreas, but both Robert and me has written code where we throw an IllegalArgumentException if we get a list with >1 item. The exception description should contain info about what happened, and what values were involved so it is easy to find the spot in the source (eg flux document) where this deviation from the Flux doc happened.